extension/src/paymentHandler/make-payment.handler.js

```diff
@@ -12,10 +12,15 @@ async function execute (paymentObject) {
     pU.createSetCustomFieldAction(c.CTP_CUSTOM_FIELD_MAKE_PAYMENT_RESPONSE, response)
   ]
 
-  if (!paymentObject.key)
+  const paymentKey = paymentObject.key
+  // ensure the key is a string, otherwise the error with "code": "InvalidJsonInput" will return by commercetools API.
+  const reference = request.reference.toString()
+  // ensure the key and new reference is different, otherwise the error with
+  // "code": "InvalidOperation", "message": "'key' has no changes." will return by commercetools API.
+  if (reference !== paymentKey)
```

Comment:
What if key is empty String? And this means user can change the key?

Reply:
The validation code below should not allow this to happen, 

https://github.com/commercetools/commercetools-adyen-integration/blob/1a0c7a715e736628ef2a648e07dcdf148e074523/extension/src/validator/validator-builder.js#L21-L29